src/Application/Processor/ProductLowestPriceBeforeDiscountProcessor.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\PriceHistoryPlugin\Application\Processor;
+
+use Sylius\Component\Channel\Context\ChannelContextInterface;
+use Sylius\PriceHistoryPlugin\Domain\Model\ChannelInterface;
+use Sylius\PriceHistoryPlugin\Domain\Model\ChannelPricingInterface;
+use Sylius\PriceHistoryPlugin\Domain\Model\ChannelPricingLogEntryInterface;
+use Sylius\PriceHistoryPlugin\Domain\Repository\ChannelPricingLogEntryRepositoryInterface;
+use Webmozart\Assert\Assert;
+
+final class ProductLowestPriceBeforeDiscountProcessor implements ProductLowestPriceBeforeDiscountProcessorInterface
+{
+    private const PERIOD_IN_DAYS = 30;
+
+    public function __construct(
+        private ChannelContextInterface $channelContext,
+        private ChannelPricingLogEntryRepositoryInterface $channelPricingLogEntryRepository,
+    ) {
+    }
+
+    public function process(ChannelPricingInterface $channelPricing): void
+    {
+        /** @var ChannelInterface $channel */
+        $channel = $this->channelContext->getChannel();
+
+        if ($channel->isLowestPriceForDiscountedProductsVisible() === false) {
+            $channelPricing->setLowestPriceBeforeDiscount(null);
+
+            return;
+        }
+
+        if (!$this->thereIsADiscount($channelPricing)) {
+            $channelPricing->setLowestPriceBeforeDiscount(null);
+
+            return;
+        }
+
+        $mostRecentLogEntry = $this->getMostRecentLogEntry($channelPricing);
+        $logEntryWithTheLowestPriceInPeriod = $this->getLogEntryWithTheLowestPriceInPeriod($mostRecentLogEntry);
+
+        if ($logEntryWithTheLowestPriceInPeriod !== null) {
+            $channelPricing->setLowestPriceBeforeDiscount($logEntryWithTheLowestPriceInPeriod->getPrice());
+
+            return;
+        }
+
+        $lowestPriceBeyondPeriod = $this->getTheMostRecentLogEntryBeyondPeriod($mostRecentLogEntry);
+
+        if ($lowestPriceBeyondPeriod === null) {
+            $channelPricing->setLowestPriceBeforeDiscount(null);
+
+            return;
+        }
+
+        if ($lowestPriceBeyondPeriod->getOriginalPrice() === null) {
+            $channelPricing->setLowestPriceBeforeDiscount(null);
+
+            return;
+        }
```

Comment:
Why?

Reply:
The test needs to be adjusted, since there is a discount the information should be shown and due to the timeframe it should be $42